### ISSUE TITLE
WCAG violation (4.1.1 and 4.1.2) with nested `button`| `link` roles

### ISSUE THREAD

**Issue by sun-mota** [AlaskaAirlines/auro-hyperlink#262]
### What browser client(s) are you seeing the issue on?

- [x] Firefox
- [x] Chrome
- [x] Safari
- [x] Microsoft Edge

### What devices platform(s) are you seeing the issue on?

- [x] Desktop
- [x] Mobile

### Steps to reproduce

![Image](https://github.com/user-attachments/assets/9093f0a6-8048-4c18-bab8-e2232f128340)

Assigning role to auro-hyperlink creates `nested interactive controls` issue

### Expected behavior

No interactive element inside or an interactive element. 

### Reproducible on the Auro doc site

- [x] I am able to reproduce this issue on the [Auro doc site](https://auro.alaskaair.com).
